### DISCUSSION TITLE
インライン数式の表示がおかしい

### DISCUSSION THREAD

**Discussion by tomtkg** [increments/qiita-discussions#498] [要望・アイデア・不具合]
インライン数式の表示がおかしいです．
前後の日本語より上の位置にインライン数式が表示されています．
日本語とインライン数式の高さは，同じであるべきだと思います．

![image](https://github.com/increments/qiita-discussions/assets/39430259/def7dffe-e9cf-4361-88a7-49388f2890ff)
次に，$D$を計算します．（列ベクトル$C$の各要素を$+H(=4)$します．）

添付画像は一例ですが，明らかに表示位置に改善の余地があると思います．

**Comment by tsubolabo** [comment]
この問題は、「数式に横スクロールバーが表示される」（#346）の付帯コメントで @hahho さんが示唆した css の overflow-x や overflow-y の設定変更により、 2022-12-15 に改良完了したはずのものです。

早々とクローズにしてしまっているので、Qiita内の関係者の目にも留まらないようになり、別のどなたかが元に戻してしまったのではありませんか？

**Reply by tsubolabo** [reply]
少し前まで、MathJax は ver.2 だったように記憶していますが、今は ver.3 のようです。いつから変更されたのですか？

**Comment by tomtkg** [comment]
https://blog.qiita.com/update-math-jax-v3/
2023/7/6の変更（MathJax v2.6.0 → v3.2.2）により発生した不具合だと思います．

[@tsubolab](https://github.com/tsubolabo)さんのご指摘の通りだと思います．

動作確認や検証が不十分なままMathJaxの更新作業をしてしまったのではないかと愚考します．
css変更で正常な表示状態できるなら，早期に対応を願いしたいです．

**Comment by kyntk** [comment]
@tomtkg さん ご報告いただきありがとうございます。
おっしゃるとおり、バージョン3への変更に伴うMathJaxの仕様変更が原因となります。
仕様が変わったため、バージョン2と同様の状態にスタイルを当てるのは難しいのですが、なんらか調整できないか検討してみます :pray: 

@tsubolabo さん 補足いただきありがとうございます。
@tomtkg さんにも補足いただきましたが、2023/7/6にバージョンアップを行った影響となります。

**Reply by tsubolabo** [reply]
@tomtkg  さんにご紹介いただいた下記ブログの内容についてコメントします。

>2023/7/6付け Qiita Blog
>「数式を描画するライブラリのバージョンアップを行います」

**バージョンアップによる効果**

エディタでのちらつきが解消されます
→　そうでしたか。動画にあるようなチラツキは未経験でした。

Qiita Discussionsにてご報告いただいていたバグも含めて、様々なバグが改善、解消されます
→　私の Discussions が２件例示されていますが、実際に改善されているのは最初の１件（#425）だけです。「様々なバグが改善、解消」は言い過ぎのような・・・。

**非互換が発生する記法**

\\\\による改行が使用できなくなります
→　これは多用している（Qiita内で実際に使用するときは\\\\ではなく\\\\\\\\）ので狼狽えました。チェックしたところ、従来の記述のままでも、特に何の問題もなく改行されています。私が考えている改行とは別の話なのでしょうか？

**Reply by kyntk** [reply]
@tsubolabo 
> 様々なバグが改善、解消

このバグが指しているのはQiita単体の話ではなく、MathJax自体の話です :pray: 
MathJaxのバージョン3でSafariでのバグを含めて様々な改善やバグ修正がされております。
詳細は下記リリースノートやドキュメントでご確認いただけますと幸いです。
https://github.com/mathjax/MathJax-src/releases
https://docs.mathjax.org/en/latest/upgrading/whats-new-3.0.html など

> 従来の記述のままでも、特に何の問題もなく改行されています。

下記のMathJaxのIssue上でもやりとりされている通り、すべての改行が使えなくなるわけではなく、問題なく使えるものもあります。
エディタにはプレビュー機能がありますので、投稿時に改行が使えない場合のみ対応いただければと思います :bow:
https://github.com/mathjax/MathJax/issues/2312

**Comment by kyntk** [comment]
@tomtkg お待たせしました 🙇 
完全に解決するのは難しそうだったのですが、少し改善を加えてみました。

お使いのブラウザによっては改善が見られない可能性もありますが、いかがでしょうか?
ご確認いただけますと幸いです。:pray:

**Reply by tomtkg** [reply]
確認いたしました．
完全に解決したとは申し上げられませんが，表示が明らかに改善されております．
早期のご対応，誠にありがとうございます．

まだ日本語とインライン数式の高さが合っていない部分がございますので，
引き続きご対応のほど，よろしくお願いいたしします．

**Reply by kyntk** [reply]
@tomtkg ご確認いただきありがとうございます！

> 完全に解決したとは申し上げられませんが，表示が明らかに改善されております．
> 早期のご対応，誠にありがとうございます．

明らかに改善されているとのことで、よかったです :sparkles: :sparkles:
そう言っていただけると、我々としてもとても嬉しいです :two_hearts: 

> まだ日本語とインライン数式の高さが合っていない部分

MathJaxの今後のバージョンアップ次第でまた変化する可能性もありますが、引き続き確認してまいります :bow:
